packages/rpc-methods/src/permitted/requestSnaps.ts

```diff
@@ -61,30 +63,72 @@ export type RequestSnapsHooks = {
 };
 
 /**
- * Checks whether existing permissions satisfy the requested permissions
- *
- * Note: Currently, we don't compare caveats, if any caveats are requested, we always return false.
+ * Checks whether an origin has existing `wallet_snap` permission and
+ * whether or not it has the requested snapIds caveat.
  *
  * @param existingPermissions - The existing permissions for the origin.
- * @param requestedPermissions - The requested permissions for the origin.
- * @returns True if the existing permissions satisfy the requested permissions, otherwise false.
+ * @param requestedSnaps - The requested snaps.
+ * @returns True if the existing permissions satisfy the requested snaps, otherwise false.
  */
-function hasPermissions(
+export function hasRequestedSnaps(
   existingPermissions: Record<string, PermissionConstraint>,
-  requestedPermissions: RequestedPermissions,
+  requestedSnaps: Record<string, unknown>,
 ): boolean {
-  return Object.entries(requestedPermissions).every(
-    ([target, requestedPermission]) => {
-      if (
-        requestedPermission?.caveats &&
-        requestedPermission.caveats.length > 0
-      ) {
-        return false;
-      }
-
-      return hasProperty(existingPermissions, target);
+  const snapIdCaveat = existingPermissions.wallet_snap?.caveats?.find(
```

Comment:
Would it make sense to use `WALLET_SNAP_PERMISSION_KEY` for this?

```suggestion
  const snapIdCaveat = existingPermissions[WALLET_SNAP_PERMISSION_KEY]?.caveats?.find(
```

Reply:
Good catch!